### PR TITLE
 treewide: convert patchPhase to postPatch

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/emulators/wine/builder-wow.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ preFlags="${configureFlags}"
 
 unpackPhase
 cd $TMP/$sourceRoot
-patchPhase
+postPatch
```
Comment (lukegb) [inline pkgs/applications/emulators/wine/builder-wow.sh:8]: This is a builder, so it shouldn't have patchPhase swapped for postPatch.

File: pkgs/applications/editors/sublime/4/common.nix [hunk 2]
```diff
@@ -29,7 +29,7 @@ in let
     nativeBuildInputs = [ zip unzip makeWrapper wrapGAppsHook ];
 
     # make exec.py in Default.sublime-package use own bash with an LD_PRELOAD instead of "/bin/bash"
-    patchPhase = ''
+    postPatch = ''
```
Comment (lukegb) [inline pkgs/applications/editors/sublime/4/common.nix:32]: This shouldn't run the hooks if it isn't a phase.

File: pkgs/applications/editors/sublime/3/common.nix [hunk 3]
```diff
@@ -43,7 +43,7 @@ in let
     nativeBuildInputs = [ zip unzip makeWrapper wrapGAppsHook ];
 
     # make exec.py in Default.sublime-package use own bash with an LD_PRELOAD instead of "/bin/bash"
-    patchPhase = ''
+    postPatch = ''
```
Comment (lukegb) [inline pkgs/applications/editors/sublime/3/common.nix:46]: This shouldn't run the hooks if it isn't a phase.

File: pkgs/development/haskell-modules/generic-builder.nix [hunk 4]
```diff
@@ -65,7 +65,7 @@ in
 , broken ? false
 , preCompileBuildDriver ? null, postCompileBuildDriver ? null
 , preUnpack ? null, postUnpack ? null
-, patches ? null, patchPhase ? null, prePatch ? "", postPatch ? ""
+, patches ? null, postPatch ? null, prePatch ? "", postPatch ? ""
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/generic-builder.nix:68]: What are you trying to do here?  This doesn't make sense and will likely not evaluate.

File: pkgs/build-support/release/maven-build.nix [hunk 5]
```diff
@@ -17,7 +17,7 @@ in
 
 stdenv.mkDerivation ( {
   inherit name src;
-  phases = "setupPhase unpackPhase patchPhase mvnCompile ${if doTestCompile then "mvnTestCompile mvnTestJar" else ""} ${if doTest then "mvnTest" else ""} ${if doJavadoc then "mvnJavadoc" else ""} ${if doCheckstyle then "mvnCheckstyle" else ""} mvnJar mvnAssembly mvnRelease finalPhase";
+  phases = "setupPhase unpackPhase postPatch mvnCompile ${if doTestCompile then "mvnTestCompile mvnTestJar" else ""} ${if doTest then "mvnTest" else ""} ${if doJavadoc then "mvnJavadoc" else ""} ${if doCheckstyle then "mvnCheckstyle" else ""} mvnJar mvnAssembly mvnRelease finalPhase";
```
Comment (sternenseemann) [inline pkgs/build-support/release/maven-build.nix:20]: Wrong.

File: pkgs/build-support/rust/hooks/cargo-setup-hook.sh [hunk 6]
```diff
@@ -34,8 +34,8 @@ EOF
 }
 
 # After unpacking and applying patches, check that the Cargo.lock matches our
-# src package. Note that we do this after the patchPhase, because the
-# patchPhase may create the Cargo.lock if upstream has not shipped one.
+# src package. Note that we do this after the postPatch, because the
+# postPatch may create the Cargo.lock if upstream has not shipped one.
```
Comment (sternenseemann) [inline pkgs/build-support/rust/hooks/cargo-setup-hook.sh:38]: This comment was accurate before.

File: pkgs/build-support/src-only/default.nix [hunk 7]
```diff
@@ -17,5 +17,5 @@ stdenv.mkDerivation (args // {
   installPhase = "cp -r . $out";
   outputs = [ "out" ];
   separateDebugInfo = false;
-  phases = ["unpackPhase" "patchPhase" "installPhase"];
+  phases = ["unpackPhase" "postPatch" "installPhase"];
```
Comment (sternenseemann) [inline pkgs/build-support/src-only/default.nix:20]: Wrong.

File: pkgs/build-support/trivial-builders.nix [hunk 8]
```diff
@@ -769,7 +769,7 @@ rec {
       inherit name src patches postPatch;
       preferLocalBuild = true;
       allowSubstitutes = false;
-      phases = "unpackPhase patchPhase installPhase";
+      phases = "unpackPhase postPatch installPhase";
```
Comment (sternenseemann) [inline pkgs/build-support/trivial-builders.nix:772]: Wrong.

File: pkgs/development/haskell-modules/make-package-set.nix [hunk 9]
```diff
@@ -576,7 +576,7 @@ in package-set { inherit pkgs lib callPackage; } self // {
         } ''
         unpackPhase
         cd "''${sourceRoot:-.}"
-        patchPhase
+        postPatch
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/make-package-set.nix:579]: This runs the `patchPhase` deliberately.

File: pkgs/development/interpreters/erlang/generic-builder.nix [hunk 10]
```diff
@@ -48,7 +48,7 @@
 , preUnpack ? ""
 , postUnpack ? ""
 , patches ? [ ]
-, patchPhase ? ""
+, postPatch ? ""
```
Comment (sternenseemann) [inline pkgs/development/interpreters/erlang/generic-builder.nix:51]: Wrong, eval failure. Changes to this file should be reverted.

File: pkgs/development/interpreters/octave/build-octave-package.nix [hunk 11]
```diff
@@ -21,7 +21,7 @@
 
 , dontPatch ? false
 , patches ? []
-, patchPhase ? ""
+, postPatch ? ""
```
Comment (sternenseemann) [inline pkgs/development/interpreters/octave/build-octave-package.nix:24]: Wrong, will break reverse dependencies.

File: pkgs/development/interpreters/python/mk-python-derivation.nix [hunk 12]
```diff
@@ -98,7 +98,7 @@
 , ... } @ attrs:
 
 
-# Keep extra attributes from `attrs`, e.g., `patchPhase', etc.
+# Keep extra attributes from `attrs`, e.g., `postPatch', etc.
```
Comment (sternenseemann) [inline pkgs/development/interpreters/python/mk-python-derivation.nix:101]: This comment was accurate before.

File: pkgs/development/libraries/hspell/dicts.nix [hunk 13]
```diff
@@ -2,7 +2,7 @@
 
 let
   dict = variant: a: stdenv.mkDerivation ({
-    inherit (hspell) version src patchPhase nativeBuildInputs;
+    inherit (hspell) version src postPatch nativeBuildInputs;
```
Comment (sternenseemann) [inline pkgs/development/libraries/hspell/dicts.nix:5]: Wrong.

File: pkgs/development/libraries/quickfix/disableUnitTests.patch [hunk 14]
```diff
@@ -1,5 +1,5 @@
-diff -u -r source-baseline-patchPhase/configure.ac source/configure.ac
---- source-baseline-patchPhase/configure.ac	1970-01-01 00:00:01.000000000 +0000
+diff -u -r source-baseline-postPatch/configure.ac source/configure.ac
```
Comment (sternenseemann) [inline pkgs/development/libraries/quickfix/disableUnitTests.patch:1]: This patch will certainly stop applying.

**Review by sternenseemann — CHANGES_REQUESTED** [review]
I only reviewed about half of the changed files and already found a concerning amount of obvious mistakes which leads me to believe you did a search & replace change without reviewing the actual changes.

Please redo this change and review every replaced match before submitting it.

**Comment by SuperSandro2000** [comment]
This was supposed to be a draft. Upsi.

And yeah, I did a mass replace.

**Pull request closed without merging**